packages/backend-for-frontend/test/api/purposeRouter/getConsumerPurposes.test.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import { generateToken } from "pagopa-interop-commons-test";
+import { authRole } from "pagopa-interop-commons";
+import { bffApi } from "pagopa-interop-api-clients";
+import request from "supertest";
+import { api, services } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import { getMockBffApiPurpose } from "../../mockUtils.js";
+import {
+  agreementNotFound,
+  eServiceNotFound,
+  eserviceDescriptorNotFound,
+  tenantNotFound,
+} from "../../../src/model/errors.js";
+
+describe("API GET /consumers/purposes test", () => {
+  const mockPurposes: bffApi.Purposes = {
+    results: [getMockBffApiPurpose(), getMockBffApiPurpose()],
+    pagination: { offset: 0, limit: 10, totalCount: 20 },
+  };
+  const defaultQuery = {
+    offset: 0,
+    limit: 5,
+  };
+
+  beforeEach(() => {
+    services.purposeService.getConsumerPurposes = vi
+      .fn()
+      .mockResolvedValue(mockPurposes);
+  });
+
+  const makeRequest = async (token: string, query: object = defaultQuery) =>
```

Comment:
For request bodies, I used the specific types. For `query` here I left `object` because I didn't find a compact way to refer to the correct type.

`QueryParametersByAlias<typeof bffApi.purposesApi.api, "getConsumerPurposes">` returns the type but has all parameters required. Elsewhere in the code I see those query parameter types used with `Partial` to restore optionality, but this loses the fact that offset and limit are instead required. Considering also that most tests are for invalid parameters, so even using the correct type would require many casts, I left `object` here.